torch/csrc/utils/python_symnode.h

```diff
@@ -95,6 +106,23 @@ class PythonSymNodeImpl : public c10::SymNodeImpl {
     return getPyObj().attr("is_bool")().is(py::handle(Py_True));
   }
 
+  bool is_singleton() const override {
+    return is_singleton_;
+  }
+
+  c10::TensorImpl* singleton_vec() const override {
+    py::gil_scoped_acquire acquire;
+    return getPyObj()
+        .attr("singleton_vec")()
+        .cast<at::Tensor>()
+        .unsafeGetTensorImpl();
```

Comment:
This is pretty unsafe because the Python code could have easily generated a temporary Tensor (at which point it immediately becomes dead)

Reply:
Is this code even live?

Reply:
Nope, good catch - I'm going to leave it and have it error with a nicer error message telling people reaching this is unexpected.